### PR TITLE
adding 64k bootloader offset to the stm32f407 mcu

### PR DESCRIPTION
**Pull request by raptordemon**
adding 64k bootloader offset to the stm32f407 mcu this allows for it to be built for the Mingda Magician X printer and the example cfg for the magician x

### CODE REVIEW

**Comment by raptordemon** [comment]
added 64k bootloader to the stm32f407 and the initial config for the mingda magician x series of printers. they have a custom bootloader which is 64k

**Comment by raptordemon** [comment]
adding 64k bootloader offset to the stm32f407 mcu this allows for it to be built for the Mingda Magician X printer and the example cfg for the magician x

**Comment by raptordemon** [comment]
module: adding support for the mingda magician x printer. it needs a 64k bootloader offset on the stm32f407 mcu

changed the kconifg in stm32 to allow for a 64k bootloader option for the mingda magician x as they use a custom bootloader for some reason.

i have also added a config file for this printer to help new users get started.

Signed-off-by: David Ingram raptordemon@gmail.com

**Comment by raptordemon** [comment]
thanks James is that what i needed to do?

**Comment by raptordemon** [comment]
Ok I will fix these too, thanks

**Comment by raptordemon** [comment]
Hello i have fixed the requested changes. pls let me knw if i have missed anything

thanks
Raptor

File: config/printer-Mingda-Magician-X-2023.cfg [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+# The "make flash" command does not work on the Mingda. Instead,
+# after running "make", copy the generated "out/klipper.bin" file to a
+# file named "firmware.bin" on an SD card and then restart the Printer
+# with that SD card.
+
+# See docs/Config_Reference.md for a description of parameters.
+#stm32 f407 64kb bootloader and 8mhz clock usb a
+
+[include macros.cfg]
+[include mainsail.cfg]
```
Comment (JamesH1978) [inline config/printer-Mingda-Magician-X-2023.cfg:14]: please do not include cfg files that an end user may not have or need

Thanks
James

File: config/printer-Mingda-Magician-X-2023.cfg [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+[filament_switch_sensor toolhead_runout]
+pause_on_runout: FALSE
+runout_gcode: {action_respond_info("RUNOUT: Toolhead Filament runout")}
+insert_gcode: {action_respond_info("RUNOUT: Toolhead Filament inserted")}
+switch_pin: ^PG1
+
+
+
+[neopixel my_neopixel]
```
Comment (JamesH1978) [inline config/printer-Mingda-Magician-X-2023.cfg:187]: Are these stock attached to the printer?

Thanks
James

File: config/printer-Mingda-Magician-X-2023.cfg [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+kinematics: cartesian
+max_velocity: 200
+max_accel: 800
+max_z_velocity: 4
+max_z_accel: 100
+
+[temperature_sensor mcu_temp]
+sensor_type: temperature_mcu
+
+[filament_switch_sensor toolhead_runout]
```
Comment (JamesH1978) [inline config/printer-Mingda-Magician-X-2023.cfg:179]: Is this a stock feature?

Thanks
James

**Review by JamesH1978 — CHANGES_REQUESTED** [review]
I have made some comments

You also need to add printer regression tests as per point 4 https://github.com/Klipper3d/klipper/blob/master/docs/Example_Configs.md

Thanks
James

**Pull request closed without merging**